### PR TITLE
Wowza

### PR DESCRIPTION
**Pull request by iiAzui**
> Thanks for contributing!
> Provide a description of your changes below and a general summary in the title.
> Please look at the following checklist to ensure that your PR can be accepted quickly:
>
> Once you've read over these comments, you should delete them.

## Description

> Describe your changes in a sentence or two.

This PR...

> Which issue does this close?
> If none, quickly write up a more detailed issue so any non-development-related discussion can go there.

_Closes: #issue_

---

## Type of Change

> Delete all non-applicable categories.

- ✨ New feature
- 🐛 Bug fix
- 💥 Breaking change
- 📝 Documentation
- ♻️ Code refactor
- 🎨 Style
- ⚡️ Performance improvements
- 🧹 Reduce tech debt
- 🔥 Remove obsolete, unused things
- 🧪 Test
- 💚 CI
- 🛠️ Build configuration change
- ⏪️ Revert
- 🗑️ Chore

## Checklist

> Before submitting the PR, please make sure you do the following

- [ ] Read the Contributing Guidelines.
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Fill out this template.
- [ ] Log your hours.
- [ ] Check that commits follow the Angular commit convention, more or less.
- [ ] Ideally, include relevant tests that fail without this PR but pass with it (if applicable).

## Tested on

> What device(s) have you tested this on?
> Delete all non-applicable examples.

- macOS 14
- Ubuntu 20.04
- Windows 10
- etc.

### CODE REVIEW

**Comment by iiAzui** [comment]
looks good

**Review by lishaduck — APPROVED** [review]
Fine.

File: addons/godot-git-plugin/win64/~libgit_plugin.windows.editor.x86_64.dll
Comment (lishaduck) [inline addons/godot-git-plugin/win64/~libgit_plugin.windows.editor.x86_64.dll:1]: Why?

File: game/characters/wizard/wizard.gd [hunk 1]
```diff
@@ -133,7 +133,9 @@ func firing() -> void:
 
 func fire(num: int) -> void:
 	var angle := self.global_position.angle_to_point(get_global_mouse_position())
-	self.did_fire.emit(num, angle, self.global_position)
+	# gdlint:ignore = private-method-call
+	level._on_wizard_did_fire(num, angle, Vector2(self.global_position.x - 10 ,self.global_position.y))
+
```
Comment (lishaduck) [inline game/characters/wizard/wizard.gd:138]: Why?

File: game/levels/level.gd [hunk 2]
```diff
@@ -47,12 +47,16 @@ func _on_wizard_did_fire(ball: int, direction: float, location: Vector2) -> void
 	spawned.add_to_group("number_balls")
 
 
-func _on_anti_math_juice_poisoned(amount: int) -> void:
+func _on_anti_math_juice_poisoned() -> void:
 	self.wizard.position = self.spawn.global_position
-	wizard.on_anti_math_juice_poisoned(amount)
```
Comment (lishaduck) [inline game/levels/level.gd:52]: Why'd you remove this functionality?

File: game/levels/level.gd [hunk 3]
```diff
@@ -47,12 +47,16 @@ func _on_wizard_did_fire(ball: int, direction: float, location: Vector2) -> void
...
-
-
-func _on_wizard_mana_updated(mana: int) -> void:
```
Comment (lishaduck) [inline game/levels/level.gd:55]: Why'd this get renamed?

File: game/levels/tutorial.tscn [hunk 4]
```diff
@@ -108,14 +108,13 @@ difficulty = 0
...
-offset_right = 2300.0
-offset_bottom = 342.0
+offset_left = 1846.0
+offset_top = 262.0
+offset_right = 2335.0
+offset_bottom = 363.0
 text = "Getting an anwser wrong uses up one of your mana!
 If you run out of mana, the game ends!
-However, defeating a math rock scores points.
-Solving problems also recharges and gives you more mana."
```
Comment (lishaduck) [inline game/levels/tutorial.tscn:118]: Fair enough.

File: game/levels/tutorial.tscn [hunk 5]
```diff
@@ -108,14 +108,13 @@ difficulty = 0
...
+"
```
Comment (lishaduck) [inline game/levels/tutorial.tscn:117]: This " should be on the line above.

File: game/scenes/anti_math_juice/Screenshot 2024-02-28 222622.png
Comment (lishaduck) [inline game/scenes/anti_math_juice/Screenshot 2024-02-28 222622.png:1]: Did this end up being used?

File: game/scenes/anti_math_juice/anti_math_juice.gd [hunk 6]
```diff
@@ -7,6 +7,11 @@ signal poisoned(amount: int)
 func _ready() -> void:
 	self.add_to_group("anti_math_juices")
 
-
 func _on_body_exited(_body: Node2D) -> void:
-	self.poisoned.emit(5)
+
+	if _body.name == "Wizard":
+	# gdlint:ignore = private-method-call
+		wizard._on_anti_math_juice_poisoned(5)
```
Comment (lishaduck) [inline game/scenes/anti_math_juice/anti_math_juice.gd:14]: Why do you hate signals so much?

File: game/scenes/anti_math_juice/anti_math_juice.gd [hunk 7]
```diff
@@ -7,6 +7,11 @@ signal poisoned(amount: int)
...
+		if self.name == "RisingAntiMathJuice":
```
Comment (lishaduck) [inline game/scenes/anti_math_juice/anti_math_juice.gd:15]: Why not this?

```suggestion
		if self == RisingAntiMathJuice:
```

File: game/scenes/checkpoint/checkpoint.gd [hunk 8]
```diff
@@ -4,6 +4,7 @@ extends Area2D
 @onready var spawn: Marker2D = %Spawn
 
 
-func _on_body_entered(body: Node2D) -> void:
-	if body is Wizard:
-		self.spawn.position = self.position
+func _on_body_entered(_body: Node2D) -> void:
```
Comment (lishaduck) [inline game/scenes/checkpoint/checkpoint.gd:7]: This parameter is used.
```suggestion
func _on_body_entered(body: Node2D) -> void:
```

File: game/spawnables/quick_math/quick_math.gd [hunk 9]
```diff
@@ -12,11 +12,22 @@ signal teleported(position: Vector2)
...
+	#var num: int = randi() % 9
+	self.quick_math_label.text = question
+	quick_math_sprite.texture = load("res://assets/sprites/number_balls_blank/1b.png")
 
 
 func teleport(value: int) -> void:
-	self.teleported.emit(self.teleport_position.global_position, value == self.answer)
+	if value == answer:
+		# gdlint:ignore = private-method-call
+		wizard._on_quick_math_ball_teleported(self.teleport_position.global_position)
```
Comment (lishaduck) [inline game/spawnables/quick_math/quick_math.gd:29]: Again, what did a signal ever do to you? :)

File: game/spawnables/quick_math/quick_math.gd [hunk 10]
```diff
@@ -12,11 +12,22 @@ signal teleported(position: Vector2)
...
+
+
 func _ready() -> void:
-	self.quick_math_label.text = self.question
-	self.quick_math_sprite.texture = load("res://assets/sprites/number_balls/%s.png" % self.answer)
-	self.add_to_group("quick_maths")
+
+	#var num: int = randi() % 9
+	self.quick_math_label.text = question
+	quick_math_sprite.texture = load("res://assets/sprites/number_balls_blank/1b.png")
```
Comment (lishaduck) [inline game/spawnables/quick_math/quick_math.gd:23]: It's always the same ball now. Is that intentional?

**Review by lishaduck — COMMENTED** [review]
Just a few questions, nothing pressing.

File: game/spawnables/quick_math/quick_math.gd (hunk 10 above)
Comment (lishaduck) [reply game/spawnables/quick_math/quick_math.gd:23]: Also, this removed putting quick_math in the quick_math node group. Maybe you removed it elsewhere, but I think it's still around.

**Comment by iiAzui** [comment]
idk i just did whatever it took to push

**Comment by lishaduck** [comment]
> idk i just did whatever it took to push

Fair enough.

**Pull request merged**